rs/consensus/src/consensus/validator.rs

```diff
@@ -3526,4 +3631,114 @@ pub mod test {
             );
         });
     }
+
+    #[test]
+    fn test_create_equivocation_proof() {
+        ic_test_utilities::artifact_pool_config::with_test_pool_config(|pool_config| {
+            let subnet_members = (0..4).map(node_test_id).collect::<Vec<_>>();
+            let ValidatorAndDependencies {
+                validator,
+                state_manager,
+                mut pool,
+                ..
+            } = setup_dependencies(pool_config, &subnet_members);
+
+            state_manager
+                .get_mut()
+                .expect_latest_certified_height()
+                .return_const(Height::from(0));
+
+            // Ensure that we don't create an equivocation proof if we have
+            // two identical blocks (one validated, one unvalidated)
+            let block = pool.make_next_block();
+            pool.insert_validated(block.clone());
+            pool.insert_unvalidated(block.clone());
+            let changeset = validator.on_state_change(&PoolReader::new(&pool));
+            assert_eq!(&changeset, &[]);
```

Comment:
With the current implementation, I would have expected the second block to be validated as well, why isn't it?

Reply:
Good catch, yes. It's not validated because I didn't advance the time sufficiently. Fixed.